File: zhuzhasi/ДЗ (7)/cpp/task2.cpp
#include <iostream>
#include <string>

class Vehicle {
protected:
    int speed;

public:
    Vehicle(int speed) : speed(speed) {}

    virtual void move() const {
        std::cout << "Транспортное средство движется" << std::endl;
    }

    virtual ~Vehicle() = default;
};

class Car : public Vehicle {
private:
    std::string fuel_type;

public:
    Car(int speed, const std::string& fuel_type)
        : Vehicle(speed), fuel_type(fuel_type) {}

    void move() const override {
        std::cout << "Автомобиль мчится по дороге" << std::endl;
    }
};

class Bike : public Vehicle {
private:
    int gear_count;

public:
    Bike(int speed, int gear_count)
        : Vehicle(speed), gear_count(gear_count) {}

    void move() const override {
        std::cout << "Велосипед едет по тропинке" << std::endl;
    }
};

int main() {
    Car c(120, "бензин");
    Bike b(25, 21);

    c.move();
    b.move();

    return 0;
}

File: zhuzhasi/ДЗ (7)/cs/task2.cs
using System;

class Vehicle
{
    protected int speed;

    public Vehicle(int speed)
    {
        this.speed = speed;
    }

    public virtual void Move()
    {
        Console.WriteLine("Транспортное средство движется");
    }
}

class Car : Vehicle
{
    private string fuelType;

    public Car(int speed, string fuelType) : base(speed)
    {
        this.fuelType = fuelType;
    }

    public override void Move()
    {
        Console.WriteLine("Автомобиль мчится по дороге");
    }
}

class Bike : Vehicle
{
    private int gearCount;

    public Bike(int speed, int gearCount) : base(speed)
    {
        this.gearCount = gearCount;
    }

    public override void Move()
    {
        Console.WriteLine("Велосипед едет по тропинке");
    }
}

class Task2
{
    static void Main()
    {
        Car c = new Car(120, "бензин");
        Bike b = new Bike(25, 21);

        c.Move();
        b.Move();
    }
}
